iogt_content_migration/management/commands/load_v1_users.py

```diff
@@ -0,0 +1,387 @@
+        cur = self.db_query(sql)
+
+        for row in self.with_progress(sql, cur, 'User Migration in progress'):
+            v1_user_id = row.pop('id')
+
+            user_data = dict(row)
+            user_data.update({'has_filled_registration_survey': True})
+
+            get_user_model().objects.update_or_create(username=row['username'], defaults=user_data)
+            user = get_user_model().objects.filter(username=row['username']).first()
+            if not user:
+                user = get_user_model().objects.create(has_filled_registration_survey=True, **row)
+
+            user_groups_sql = f'select * from auth_user_groups aug ' \
+                              f'inner join auth_group ag on aug.group_id = ag.id ' \
+                              f'where user_id={v1_user_id}'
+
+            user_groups_cursor = self.db_query(user_groups_sql)
+
+            user.groups.clear()
+            for row_ in user_groups_cursor:
+                group = Group.objects.get(name=row_['name'])
+                group.user_set.add(user)
+
+            user_groups_cursor.close()
+
+    def migrate_user_groups(self):
+        self.stdout.write(self.style.SUCCESS('Starting User Groups Migration'))
+
+        sql = f'select * from auth_group'
+        cur = self.db_query(sql)
+
+        for row in cur:
+            Group.objects.get_or_create(name=row['name'])
+
+        self.stdout.write(self.style.SUCCESS('Completed User Groups Migration'))
+
+    def mark_user_registration_survey_required(self):
+        users = get_user_model().objects.filter(groups__id__in=self.registration_survey_mandatory_group_ids)
+        users.update(has_filled_registration_survey=False)
+
+    def migrate_user_comments(self):
+        self.stdout.write(self.style.SUCCESS('Starting Comment migration'))
+
+        sql = f'select * from django_comments dc inner join django_content_type dct on dc.content_type_id = dct.id'
+        cur = self.db_query(sql)
+
+        for row in self.with_progress(sql, cur, 'User comments migration in progress...'):
+            row.pop('id')
+            content_type = self.content_type_map[row['model']]
+
+            if not content_type:
+                self.stdout.write(self.style.ERROR(f'Content Type for {row["model"]} not found.'))
+                continue
+
+            try:
+                new_article = self.articles_map[row["object_pk"]]
+            except KeyError:
+                new_article
+
+            max_thread_id_comment = XtdComment.objects.filter(
+                content_type_id=content_type.id, object_pk=new_article.pk).order_by('-thread_id').first()
+
+            max_thread_id = 0
+            if max_thread_id_comment:
+                max_thread_id = max_thread_id_comment.id
+
+            XtdComment.objects.create(
+                content_type_id=content_type.id, object_pk=new_article.pk, user_name=row['user_name'],
+                user_email=row['user_email'], submit_date=row['submit_date'],
```

Comment:
When testing this, it appears that the submission date is overwritten with the current date (i.e. when you run the load_v1_users) script. This should be fixed.